soco/discovery.py

```diff
@@ -123,18 +158,19 @@ def discover(timeout=1, include_invisible=False, interface_addr=None):
         # X-RINCON-HOUSEHOLD: Sonos_7O********************R7eU
 
         if response:
-            data, addr = _sock.recvfrom(1024)
-            _LOG.debug('Received discovery response from %s: "%s"', addr, data)
-            if b"Sonos" not in data:
-                continue
-
-            # Now we have an IP, we can build a SoCo instance and query that
-            # player for the topology to find the other players. It is much
-            # more efficient to rely upon the Zone Player's ability to find
-            # the others, than to wait for query responses from them
-            # ourselves.
-            zone = config.SOCO_CLASS(addr[0])
-            if include_invisible:
-                return zone.all_zones
-            else:
-                return zone.visible_zones
+            for _sock in response:
+                data, addr = _sock.recvfrom(1024)
+                _LOG.debug(
+                    'Received discovery response from %s: "%s"', addr, data
+                )
+                if b"Sonos" in data:
+                    # Now we have an IP, we can build a SoCo instance and query
+                    # that player for the topology to find the other players.
+                    # It is much more efficient to rely upon the Zone
+                    # Player's ability to find the others, than to wait for
+                    # query responses from them ourselves.
+                    zone = config.SOCO_CLASS(addr[0])
```

Comment:
No, for two reasons:

1. The select call will return as soon as the very first response, on whatever interface, is received. So the rest of the code will never see more than one valid response, and 

2. Because of the way the SoCo class is written, no matter how many times you try to create one with the same IP address, you will only end up with one.